File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03642/p03642_num6_parsed.py
import math

def is_prime(num):
    if num < 2:
        return False
    for i in range(2, int(math.sqrt(num))+1):
        if num % i == 0:
            return False
    return True

def minimum_operations(N, x):
    max_x = max(x)
    primes = [i for i in range(2, max_x+1) if is_prime(i)]

    operations = 0
    for prime in primes:
        if all((num % prime) == 1 for num in x):
            operations += 1
    return operations

# Read input
N = int(input())
x = list(map(int, input().split()))

# Call the function and print the result
print(minimum_operations(N, x))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03642/p03642_num6_parsed.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

bool isPrime(int n) {
    if (n <= 1) {
        return false;
    }
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    int N;
    cin >> N;
    
    vector<int> cards(N);
    for (int i = 0; i < N; i++) {
        cin >> cards[i];
    }
    
    int operations = 0;
    for (int i = 0; i < N; i++) {
        if (isPrime(cards[i])) {
            operations++;
            int j = i + 1;
            while (j < N && cards[j] - cards[j-1] == 1) {
                j++;
            }
            i = j - 1;
        }
    }
    
    cout << operations << endl;
    
    return 0;
}